src/script/miniscript.h

```diff
@@ -846,6 +857,56 @@ struct Node {
         assert(false);
     }
 
+    internal::WitnessSize CalcWitnessSize() const {
+        switch (fragment) {
+            case Fragment::JUST_0: return {{}, 0};
+            case Fragment::JUST_1:
+            case Fragment::OLDER:
+            case Fragment::AFTER: return {0, {}};
+            case Fragment::PK_K: return {1 + 72, 1};
+            case Fragment::PK_H: return {1 + 72 + 1 + 33, 1 + 1 + 33};
+            case Fragment::SHA256:
+            case Fragment::RIPEMD160:
+            case Fragment::HASH256:
+            case Fragment::HASH160: return {1 + 32, {}};
+            case Fragment::ANDOR: {
+                const auto sat{(subs[0]->ws.sat + subs[1]->ws.sat) | (subs[0]->ws.dsat + subs[2]->ws.sat)};
+                const auto dsat{subs[0]->ws.dsat + subs[2]->ws.dsat};
+                return {sat, dsat};
+            }
+            case Fragment::AND_V: return {subs[0]->ws.sat + subs[1]->ws.sat, {}};
+            case Fragment::AND_B: return {subs[0]->ws.sat + subs[1]->ws.sat, subs[0]->ws.dsat + subs[1]->ws.dsat};
+            case Fragment::OR_B: {
+                const auto sat{(subs[0]->ws.dsat + subs[1]->ws.sat) | (subs[0]->ws.sat + subs[1]->ws.dsat)};
+                const auto dsat{subs[0]->ws.dsat + subs[1]->ws.dsat};
+                return {sat, dsat};
+            }
+            case Fragment::OR_C: return {subs[0]->ws.sat | (subs[0]->ws.dsat + subs[1]->ws.sat), {}};
+            case Fragment::OR_D: return {subs[0]->ws.sat | (subs[0]->ws.dsat + subs[1]->ws.sat), subs[0]->ws.dsat + subs[1]->ws.dsat};
+            case Fragment::OR_I: return {(subs[0]->ws.sat + 1 + 1) | (subs[1]->ws.sat + 1), (subs[0]->ws.dsat + 1 + 1) | (subs[1]->ws.dsat + 1)};
```

Comment:
In 3e3521bf956c4387abe4b6ce12ce55224a9e97ba "miniscript: introduce a helper to get the maximum witness size"

I don't quite follow why `subs[0]` needs an additional `+ 1` that `subs[1]` doesn't have. AFAICT, `subs[0]` just has a `1` in the witness, while `subs[1]` has a `0`, and both should be the same size.

Reply:
Because serializing a `1` as a witness stack element needs to specify a 1-byte long element and then the value of this byte (serialized `0x0101`) whereas encoding a `0` is simply the empty witness (serialized `0x00`).

Reply:
Oh right, duh.